docs/graphql/core/guides/environments.rst

```diff
@@ -0,0 +1,268 @@
+
+.. code-block:: bash
+
+    hasura migrate squash --from <version>
+
+
+Metadata
+^^^^^^^^
+
+The state of Hasura metadata is managed via snapshots of the metadata. Hasura stores this metadata to create the GraphQL API over Postgres and provide other functionalities like remote schemas, event triggers etc. 
+All the actions performed on the console, like tracking tables/views/functions, creating relationships, configuring permissions, 
+creating event triggers and remote schemas, etc. can be exported as a JSON/yaml metadata file.
+
+These snapshots can be :ref:`exported <export_hasura_metadata>` and :ref:`imported <import_hasura_metadata>` as a whole to configure Hasura to a state represented in the snapshot.
+
+The metadata directory of your Hasura project should be included in your version control system like git, 
+so that you can roll back corresponding changes later, if required.
+
+.. note::
+
+    Read more about Hasura metadata on :ref:`this docs page <manage_hasura_metadata>`.
+
+Developing and testing business logic
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Hasura lets you write business logic in a flexible way. 
+If you are comfortable writing your own GraphQL server from scratch, you can add them as a :ref:`remote schema <remote_schemas>`. 
+If you are interested in (re)using REST API endpoints, you can map GraphQL types with :ref:`actions <actions>`.
+
+Adding either a remote schema or action will become part of Hasura's metadata. 
+When you are adding either of them, you will have to give the HTTP handler / URL so that Hasura can communicate with that endpoint.
+
+Assuming that the handler is also running on your local machine, 
+you will need to give an endpoint that is accessible from inside the Docker container of Hasura.
+Typically for Mac, this will be ``host.docker.internal`` and for Linux where the containers are running in ``host`` mode, 
+it will be ``localhost``.
+
+.. note::
+
+    See :ref:`this docs page <docker_networking>` to learn more about Docker networking.
+
+Configuring handlers via environment variables
+**********************************************
+
+The recommended way to configure these handler URLs is via environment variables, 
+irrespective of the development environment (local/staging/prod).
+
+**Actions**
+
+Actions can have a base URL through ENVs, something like ``{{ACTION_BASE_URL}}``. 
+For example, if all your REST API endpoints are running in a single server, you can configure the ENV with the host name.
+
+.. code-block:: bash
+
+    {{ACTION_BASE_URL}}/createUser
+
+The ``{{ACTION_BASE_URL}}`` will typically have values like ``http://myserver.com`` or when running in localhost, 
+it will look something like ``http://localhost:3000``. All of this will be passed to the graphql-engine server as ENVs.
+
+Forwarding headers can be configured using ENV. 
+This will be applicable when you are sharing a common secret between your action handler or when passing some authorization tokens etc.
```

Comment:
leave another line after **Forwarding headers can be configured using ENV.**  so it won't be like the header and the note are on the same line.

Reply:
It's not a header. It looks like this on the page:

![image](https://user-images.githubusercontent.com/5722022/93490277-926ea200-f908-11ea-9255-c5cd9ff21d31.png)

It's best to review these PRs with the preview app that is automatically generated. Then you see how it will actually look 🙂